### PR TITLE
Update Call state and docs

### PR DESCRIPTION
**Pull request by ipavlidakis**
### 🔗 Issue Links

_Provide all JIRA tickets and/or GitHub issues related to this PR, if applicable._

### 🎯 Goal

_Describe why we are making this change._

### 📝 Summary

_Provide bullet points with the most important changes in the codebase._

### 🛠 Implementation

_Provide a detailed description of the implementation and explain your decisions if you find them relevant._

### 🎨 Showcase

_Add relevant screenshots and/or videos/gifs to easily see what this PR changes, if applicable._

|  Before  |  After  |
| -------- | ------- |
|  `img`   |  `img`  |

### 🧪 Manual Testing Notes

_Explain how this change can be tested manually, if applicable._

### ☑️ Contributor Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] This change follows zero ⚠️ policy (required)
- [x] This change should receive manual QA
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (Docusaurus, tutorial, CMS)

### 🎁 Meme

_Provide a funny gif or image that relates to your work on this pull request. (Optional)_

### CODE REVIEW

File: Sources/StreamVideo/CallState.swift [hunk 1]
```diff
@@ -180,5 +187,38 @@ public class CallState: ObservableObject {
...
     }
-    
+
+    private func didUpdate(_ participants: [CallParticipant]) {
+        var remoteParticipants: [CallParticipant] = []
+        var activeSpeakers: [CallParticipant] = []
+        var screenSharingSession: ScreenSharingSession?
+
+        for participant in participants {
+            if participant.id == streamVideo.user.id {
```
Comment (martinmitrevski) [inline Sources/StreamVideo/CallState.swift:197]: This is not good enough condition, since a participant with the same id can appear from different devices. We should change the right part of the equation to the session id.

File: Sources/StreamVideo/Models/CallParticipant.swift [hunk 2]
```diff
@@ -23,7 +23,7 @@ public struct CallParticipant: Identifiable, Sendable, Equatable {
     public var hasVideo: Bool
     /// Returns whether the participant has audio.
     public var hasAudio: Bool
-    /// Returns whether the participant is screensharing.
+    /// Returns whether the participant is screenSharing.
```
Comment (martinmitrevski) [inline Sources/StreamVideo/Models/CallParticipant.swift:26]: why this change?

File: Sources/StreamVideo/Models/ScreenSharingSession.swift [hunk 3]
```diff
@@ -0,0 +1,14 @@
+//
```
Comment (martinmitrevski) [inline Sources/StreamVideo/Models/ScreenSharingSession.swift:1]: wrong header

File: Sources/StreamVideo/Models/CallParticipant.swift (hunk 2 above)
Comment (ipavlidakis) [reply Sources/StreamVideo/Models/CallParticipant.swift:26]: A few reasons. It's 2 words and capitalising the second word letter makes sense. Xcode was presenting as a typo. Finally it's consistent with the naming of our view (ScreenSharingView).

File: Sources/StreamVideoSwiftUI/CallView/CallView.swift [hunk 4]
```diff
@@ -31,10 +31,10 @@ public struct CallView<Factory: ViewFactory>: View {
                             localVideoView
                                 .edgesIgnoringSafeArea(.top)
                                 .accessibility(identifier: "localVideoView")
-                        } else if let screensharing = viewModel.screensharingSession {
+                        } else if let screenSharingSession = viewModel.call?.state.screenSharingSession {
```
[the remark this answers is not in the document]
Comment (ipavlidakis) [reply Sources/StreamVideoSwiftUI/CallView/CallView.swift:34]: I tested it and it works as it was. Do you see any issues with it?

[the remark this answers is not in the document]
Comment (martinmitrevski) [reply Sources/StreamVideoSwiftUI/CallView/CallView.swift:34]: no, just wanted to make sure.

File: Sources/StreamVideoSwiftUI/CallViewModel.swift [hunk 5]
```diff
@@ -47,6 +47,14 @@ open class CallViewModel: ObservableObject {
                         }
                     }
                 })
+            screenSharingUpdates = call?.state.$screenSharingSession
+                .receive(on: RunLoop.main)
+                .sink(receiveValue: { [weak self] screenSharingSession in
+                    if screenSharingSession?.participant.id != self?.lastScreenSharingParticipant?.id {
+                        self?.lastLayoutChange = Date()
+                    }
+                    self?.lastScreenSharingParticipant = screenSharingSession?.participant
```
Comment (martinmitrevski) [inline Sources/StreamVideoSwiftUI/CallViewModel.swift:56]: I don't think we need this?

**Review by martinmitrevski — COMMENTED** [review]
Almost there, just one question

**Review by martinmitrevski — APPROVED** [review]
